File: ros_ws05_advance/src/dynamic_config_test/scripts/config_server.py
#! /usr/bin/env python

import rospy
from dynamic_reconfigure.server import Server
from dynamic_config_test.cfg import drConfig


def callback(config, level):
    rospy.loginfo(
        "dynamic param:%d,%.2f,%d,%s,%d",
        config.int_param,
        config.double_param,
        config.bool_param,
        config.string_param,
        config.list_param,
    )
    return config


if __name__ == "__main__":
    rospy.init_node("dynamic_param_py")
    server = Server(drConfig, callback)
    rospy.spin()

File: ros_ws05_advance/src/dynamic_config_test/src/config_server.cpp
#include "ros/ros.h"
#include "dynamic_config_test/drConfig.h"
#include "dynamic_reconfigure/server.h"

void callback(dynamic_config_test::drConfig& config, uint32_t level)
{
    ROS_INFO("dynamic data:%d,%.2f,%d,%s,%d",
    config.int_param,
    config.double_param,
    config.bool_param,
    config.string_param.c_str(),
    config.list_param);
}

int main(int argc,char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc,argv,"dr");
    dynamic_reconfigure::Server<dynamic_config_test::drConfig> server;
    dynamic_reconfigure::Server<dynamic_config_test::drConfig>::CallbackType cbtype;
    cbtype = boost::bind(&callback,_1,_2);
    server.setCallback(cbtype);
    ros::spin();
    return 0;
}